### PR TITLE
feat(client): Add `RollupConfig` to `BootInfo`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a function in the `BootInfo` module that pulls in the rollup config based on the passed L2 chain ID.

### CODE REVIEW

**Comment by clabby** [comment]
* **#252** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/252?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#251** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/251?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: bin/host/src/cli/mod.rs [hunk 1]
```diff
@@ -52,6 +52,9 @@ pub struct HostCli {
     /// Number of the L2 block that the claim is from.
     #[clap(long)]
     pub l2_block_number: u64,
+    /// The L2 chain ID.
+    #[clap(long)]
+    pub l2_chain_id: u64,
     //// Path to the genesis file.
     #[clap(long)]
     pub l2_genesis_path: PathBuf,
```
Comment (refcell) [inline bin/host/src/cli/mod.rs:60]: Can we load the genesis from the superchain registry by l2 chain id?

File: bin/host/src/cli/mod.rs [hunk 2]
```diff
@@ -52,6 +52,9 @@ pub struct HostCli {
     /// Number of the L2 block that the claim is from.
     #[clap(long)]
     pub l2_block_number: u64,
+    /// The L2 chain ID.
+    #[clap(long)]
+    pub l2_chain_id: u64,
```
Comment (refcell) [inline bin/host/src/cli/mod.rs:57]: How is this different from the `network` field above.

File: bin/host/src/kv/local.rs [hunk 3]
```diff
@@ -29,9 +29,8 @@ impl KeyValueStore for LocalKeyValueStore {
             L2_OUTPUT_ROOT_KEY => Some(self.cfg.l2_output_root.to_vec()),
             L2_CLAIM_KEY => Some(self.cfg.l2_claim.to_vec()),
             L2_CLAIM_BLOCK_NUMBER_KEY => Some(self.cfg.l2_block_number.to_be_bytes().to_vec()),
-            L2_CHAIN_ID_KEY => todo!(),
-            L2_CHAIN_CONFIG_KEY => todo!(),
-            L2_ROLLUP_CONFIG_KEY => todo!(),
+            L2_CHAIN_ID_KEY => Some(self.cfg.l2_chain_id.to_be_bytes().to_vec()),
+            L2_ROLLUP_CONFIG_KEY => unimplemented!("L2RollupConfig fetching in local store not implemented. Necessary for chain IDs without a known rollup config."),
```
Comment (refcell) [inline bin/host/src/kv/local.rs:33]: Will this pull from the SCR?

File: bin/programs/client/src/boot.rs [hunk 4]
```diff
@@ -89,7 +89,16 @@ impl BootInfo {
...
+    }
+}
 
-        Ok(Self { l1_head, l2_output_root, l2_claim, l2_claim_block, chain_id })
+/// Returns the rollup config for the given chain ID.
+fn rollup_config_from_chain_id(chain_id: u64) -> Result<RollupConfig> {
+    match chain_id {
+        10 => Ok(OP_MAINNET_CONFIG),
+        _ => anyhow::bail!("Unsupported chain ID: {}", chain_id),
     }
```
Comment (refcell) [inline bin/programs/client/src/boot.rs:103]: This should be loaded from SCR right?

**Review by refcell — COMMENTED** [review]
Some comments

**Review by refcell — APPROVED** [review]

File: bin/programs/client/src/boot.rs (hunk 4 above)
Comment (refcell) [reply bin/programs/client/src/boot.rs:103]: https://github.com/ethereum-optimism/kona/issues/261

File: bin/host/src/cli/mod.rs (hunk 1 above)
Comment (refcell) [reply bin/host/src/cli/mod.rs:60]: https://github.com/ethereum-optimism/kona/issues/262

**Pull request merged**